### PR TITLE
Add update password api

### PR DESCRIPTION
**Pull request by rishabhj1717**
- Add new api and route to update password
- Return 400 incase of incorrect user id provided

### CODE REVIEW

File: backend/routes/user.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 const express = require("express");
 const router = express.Router();
-const { createUser, getUser, getAllUsers } = require("./../controllers/user");
+const { createUser, getUser, getAllUsers, updatePassword } = require("./../controllers/user");
 
 router.get("/user", getUser);
 router.get("/users", getAllUsers);
 router.post("/user", createUser);
+router.post("user/updatePassword", updatePassword);
```
Comment (megh-khaire) [inline backend/routes/user.js]: /user/change-password

File: backend/controllers/user.js [hunk 2]
```diff
@@ -48,3 +48,31 @@ module.exports.getAllUsers = async (req, res) => {
     });
   }
 };
+
+module.exports.updatePassword = async (req, res) => {
```
Comment (megh-khaire) [inline backend/controllers/user.js:52]: Do we need to check if the new password = old password?

**Review by megh-khaire — CHANGES_REQUESTED** [review]

**Review by megh-khaire — APPROVED** [review]

File: backend/app.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ app.use(
 );
 
 // CORS
-const whitelist = ["http://localhost:5000", "http://localhost:3000"];
+const whitelist = ["http://localhost:5000", "http://localhost:3000", "http://localhost:5001"];
```
Comment (megh-khaire) [inline backend/app.js]: Why do we need this?

**Review by megh-khaire — APPROVED** [review]

File: backend/controllers/user.js (hunk 2 above)
Comment (rishabhj1717) [reply backend/controllers/user.js:52]: We could give it as a task to the students 😅

File: backend/routes/user.js (hunk 1 above)
Comment (rishabhj1717) [reply backend/routes/user.js]: Done

File: backend/app.js (hunk 3 above)
Comment (rishabhj1717) [reply backend/app.js]: I forgot to remove this. Was testing something

**Pull request merged**